File: src/WorkBC.ElasticSearch.Models/Helpers/SalaryRangeHelper.cs
using System;
using System.Collections.Generic;

namespace WorkBC.ElasticSearch.Models.Helpers
{
    public static class SalaryRangeHelper
    {
        public enum SalaryType
        {
            HOURLY = 0,
            WEEKLY = 1,
            BI_WEEKLY = 2,
            MONTHLY = 3,
            ANNUALLY = 4,
            NONE = 5
        }

        private static SalaryRange[] Hourly
        {
            get
            {
                return new[]
                {
                    new SalaryRange {Minimum = 0m, Maximum = 19.999999m},
                    new SalaryRange {Minimum = 20m, Maximum = 29.999999m},
                    new SalaryRange {Minimum = 30m, Maximum = 39.999999m},
                    new SalaryRange {Minimum = 40m, Maximum = 49.999999m},
                    new SalaryRange {Minimum = 50m, Maximum = 50000m}
                };
            }
        }

        private static SalaryRange[] Weekly
        {
            get
            {
                return new[]
                {
                    new SalaryRange {Minimum = 0m, Maximum = 799.999999m},
                    new SalaryRange {Minimum = 800m, Maximum = 1199.999999m},
                    new SalaryRange {Minimum = 1200m, Maximum = 1599.999999m},
                    new SalaryRange {Minimum = 1600m, Maximum = 1999.999999m},
                    new SalaryRange {Minimum = 2000m, Maximum = 200000m}
                };
            }
        }

        private static SalaryRange[] Biweekly
        {
            get
            {
                return new[]
                {
                    new SalaryRange {Minimum = 0m, Maximum = 1599.999999m},
                    new SalaryRange {Minimum = 1600m, Maximum = 2399.999999m},
                    new SalaryRange {Minimum = 2400m, Maximum = 3199.999999m},
                    new SalaryRange {Minimum = 3200m, Maximum = 3999.999999m},
                    new SalaryRange {Minimum = 4000m, Maximum = 400000m}
                };
            }
        }

        private static SalaryRange[] Monthly
        {
            get
            {
                return new[]
                {
                    new SalaryRange {Minimum = 0m, Maximum = 3999.999999m},
                    new SalaryRange {Minimum = 4000m, Maximum = 5999.999999m},
                    new SalaryRange {Minimum = 6000m, Maximum = 7999.999999m},
                    new SalaryRange {Minimum = 8000m, Maximum = 9999.999999m},
                    new SalaryRange {Minimum = 10000m, Maximum = 10000000m}
                };
            }
        }

        private static SalaryRange[] Annually
        {
            get
            {
                return new[]
                {
                    new SalaryRange {Minimum = 0m, Maximum = 39999.999999m},
                    new SalaryRange {Minimum = 40000m, Maximum = 59999.999999m},
                    new SalaryRange {Minimum = 60000m, Maximum = 79999.999999m},
                    new SalaryRange {Minimum = 80000m, Maximum = 99999.999999m},
                    new SalaryRange {Minimum = 100000m, Maximum = 100000000m}
                };
            }
        }

        public static KeyValuePair<string, string> GetAnnualRange(SalaryType salaryType, int bracket)
        {
            SalaryRange range;
            int multiplier;

            if (bracket > 5 || bracket < 1)
            {
                throw new IndexOutOfRangeException("Salary bracket must be between 1 and 5");
            }

            switch (salaryType)
            {
                case SalaryType.HOURLY:
                    multiplier = 2080;
                    range = Hourly[bracket - 1];
                    break;

                case SalaryType.WEEKLY:
                    multiplier = 52;
                    range = Weekly[bracket - 1];
                    break;

                case SalaryType.BI_WEEKLY:
                    multiplier = 26;
                    range = Biweekly[bracket - 1];
                    break;

                case SalaryType.MONTHLY:
                    multiplier = 12;
                    range = Monthly[bracket - 1];
                    break;

                default:
                    multiplier = 1;
                    range = Annually[bracket - 1];
                    break;
            }

            return new KeyValuePair<string, string>
            (
                (multiplier * range.Minimum).ToString("0.00"),
                (multiplier * range.Maximum -0.01m).ToString("0.00")
            );
        }

        private class SalaryRange
        {
            public decimal Minimum { get; set; }
            public decimal Maximum { get; set; }
        }
    }
}

File: src/WorkBC.Notifications.JobAlerts.V2/src/Search/SalaryRangeHelper.php
<?php

declare(strict_types=1);

namespace WorkBC\JobAlertNotifier\Search;

/**
 * Port of WorkBC.ElasticSearch.Models.Helpers.SalaryRangeHelper.
 *
 * The C# version does decimal math: annualMin = multiplier * min,
 * annualMax = multiplier * max - 0.01m, both rendered with ToString("0.00").
 * PHP floats would drift on the *.999999 bracket bounds, so the math here is
 * done in integer millionths of a dollar and rounded to cents at the end.
 */
final class SalaryRangeHelper
{
    public const HOURLY = 0;
    public const WEEKLY = 1;
    public const BI_WEEKLY = 2;
    public const MONTHLY = 3;
    public const ANNUALLY = 4;
    public const NONE = 5;

    /** Bracket bounds in millionths of a dollar, per salary type. */
    private const RANGES = [
        self::HOURLY => [
            [0, 19999999],
            [20000000, 29999999],
            [30000000, 39999999],
            [40000000, 49999999],
            [50000000, 50000000000],
        ],
        self::WEEKLY => [
            [0, 799999999],
            [800000000, 1199999999],
            [1200000000, 1599999999],
            [1600000000, 1999999999],
            [2000000000, 200000000000],
        ],
        self::BI_WEEKLY => [
            [0, 1599999999],
            [1600000000, 2399999999],
            [2400000000, 3199999999],
            [3200000000, 3999999999],
            [4000000000, 400000000000],
        ],
        self::MONTHLY => [
            [0, 3999999999],
            [4000000000, 5999999999],
            [6000000000, 7999999999],
            [8000000000, 9999999999],
            [10000000000, 10000000000000],
        ],
        self::ANNUALLY => [
            [0, 39999999999],
            [40000000000, 59999999999],
            [60000000000, 79999999999],
            [80000000000, 99999999999],
            [100000000000, 100000000000000],
        ],
    ];

    private const MULTIPLIERS = [
        self::HOURLY => 2080,
        self::WEEKLY => 52,
        self::BI_WEEKLY => 26,
        self::MONTHLY => 12,
    ];

    /**
     * Returns [minAnnual, maxAnnual] as "0.00"-formatted strings, matching
     * the C# KeyValuePair<string,string> exactly.
     *
     * @return array{0: string, 1: string}
     */
    public static function getAnnualRange(int $salaryType, int $bracket): array
    {
        if ($bracket > 5 || $bracket < 1) {
            throw new \OutOfRangeException('Salary bracket must be between 1 and 5');
        }

        // Unknown/NONE/ANNUALLY all fall back to the annual table with
        // multiplier 1, matching the C# switch default.
        $multiplier = self::MULTIPLIERS[$salaryType] ?? 1;
        $ranges = self::RANGES[$salaryType] ?? self::RANGES[self::ANNUALLY];
        [$minMicro, $maxMicro] = $ranges[$bracket - 1];

        // maxAnnual = multiplier * max - 0.01 (0.01 dollars = 10,000 millionths)
        return [
            self::microToMoney($multiplier * $minMicro),
            self::microToMoney($multiplier * $maxMicro - 10000),
        ];
    }

    /** Millionths of a dollar → "0.00" string (round half away from zero, like decimal.ToString). */
    private static function microToMoney(int $micro): string
    {
        $cents = intdiv($micro, 10000);
        if (($micro % 10000) * 2 >= 10000) {
            $cents++;
        }
        return sprintf('%d.%02d', intdiv($cents, 100), $cents % 100);
    }
}
